source/interface/azure_iot_mqtt.h

```diff
@@ -0,0 +1,380 @@
+/* Copyright (c) Microsoft Corporation. All rights reserved. */
+/* SPDX-License-Identifier: MIT */
+
+/**
+ * @file azure_iot_mqtt.h
+ *
+ * @brief Definition for the Azure IoT MQTT.
```

Comment:
Also could we add a comment if this is the abstraction layer that this is subject to change, since we don't have enough data on MQTT stacks to know.